test/components/componentTests.ts

```diff
@@ -1,74 +1,87 @@
+      assert.strictEqual(c.anchor(svg), c, "setter returns calling object");
+      assert.strictEqual(Plottable.Utils.DOM.boundingSVG(<SVGElement> c.content().node()),
+        svg.node(), "component child of svg");
+      c.destroy();
+      svg.remove();
+    });
+
+    it("adds default selections in the correct order", () => {
+      c.anchor(svg);
+      assert.isFalse(c.foreground().empty(), "foreground exists in the DOM");
+      assert.isFalse(c.background().empty(), "background exists in the DOM");
+      assert.isFalse(c.content().empty(), "content exists in the DOM");
+      assert.isFalse(c.background().select(".background-fill").empty(), "background fill container exists in the DOM");
+
+      let componentElement = svg.select(".component");
+      let containers = componentElement.selectAll("g");
+      assert.strictEqual(containers[0][0], c.background().node(), "background at the back");
+      assert.strictEqual(containers[0][1], c.content().node(), "content at the middle");
+      assert.strictEqual(containers[0][2], c.foreground().node(), "foreground at the front");
+      assert.strictEqual(containers[0][3], componentElement.select(".box-container").node(), "boxes at front of foreground");
+      c.destroy();
+      svg.remove();
+    });
+
+    it("classes the input with 'plottable' if it is an svg", () => {
       c.anchor(svg);
-      assert.strictEqual((<any> c)._element.node(), svg.select("g").node(), "the component anchored to a <g> beneath the <svg>");
       assert.isTrue(svg.classed("plottable"), "<svg> was given \"plottable\" CSS class");
-      const computedStyle = window.getComputedStyle(<Element>svg.node());
+      c.destroy();
+      svg.remove();
+    });
+
+    it("allows mouse events to operate if the component is visible", () => {
+      c.anchor(svg);
+      let computedStyle = window.getComputedStyle(<Element>svg.node());
       assert.strictEqual(computedStyle.pointerEvents.toLowerCase(), "visiblefill",
         "\"pointer-events\" style set to \"visiblefill\"");
+      c.destroy();
       svg.remove();
     });
 
-    it("can re-anchor() to a different element", () => {
+    it("can switch which element it is anchored to", () => {
       c.anchor(svg);
 
       let svg2 = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
       c.anchor(svg2);
-      assert.strictEqual((<any> c)._element.node(), svg2.select("g").node(), "the component re-achored under the second <svg>");
-      assert.isTrue(svg2.classed("plottable"), "second <svg> was given \"plottable\" CSS class");
-      const computedStyle = window.getComputedStyle(<Element>svg2.node());
-      assert.strictEqual(computedStyle.pointerEvents.toLowerCase(), "visiblefill",
-        "second <svg>'s \"pointer-events\" style set to \"visiblefill\"");
-
-      svg.remove();
+      assert.strictEqual(Plottable.Utils.DOM.boundingSVG(<SVGElement> c.content().node()),
+        svg2.node(), "component child of new svg");
```

Comment:
component is a child of new svg

can you also check if component is still a child of original svg?